tests/test_gbs.py

```diff
@@ -581,7 +580,7 @@ def test_probability_non_analytic_all_wires(self):
         """Test that the probability method in non-analytic mode returns the expected dictionary
         mapping from samples to probabilities when a fixed set of pre-generated samples are
         input. The lexicographic order of the keys in the returned dictionary is also checked."""
-        dev = qml.device("strawberryfields.gbs", wires=4, cutoff_dim=3, analytic=False)
+        dev = qml.device("strawberryfields.gbs", wires=4, cutoff_dim=3, shots=4)
```

Comment:
E.g. here, `shots=4` but `len(samples) = 10`. It probably works fine because `shots` is just ignored aside from establishing that we're in non-analytic mode :thinking:

Reply:
Yep - I wasn't sure whether this was a 'bug' or not, since this was the same behaviour _before_ the shots refactor - hence why I left it as is.